cmd/otelcorecol/go.mod

```diff
@@ -2,7 +2,7 @@
 
 module go.opentelemetry.io/collector/cmd/otelcorecol
 
-go 1.22.0
```

Comment:
i don't know if we want this change. wdyt @mx-psi

Reply:
Yeah I don't think we want it and I am suprised we have this here. I spoke with @pellared about this recently, maybe he has more context?

Reply:
i opened an issue here for it: https://github.com/open-telemetry/opentelemetry-go/issues/5969

it's possible we can't do anything about it (because of the other deps that updated)